java/memory/src/main/java/io/netty/buffer/ArrowBuf.java

```diff
@@ -1204,4 +1204,51 @@ public ArrowBuf reallocIfNeeded(final int size) {
               "Realloc is only available in the context of operator's UDFs");
     }
   }
+
+  /**
+   * Following are wrapper methods to keep this backward compatible.
+   */
+  @Deprecated
+  public void release() {
+    referenceManager.release();
+  }
+
+  @Deprecated
+  public void release(int decrement) {
+    referenceManager.release(decrement);
+  }
+
+  @Deprecated
+  public void retain() {
+    referenceManager.retain();
+  }
+
+  @Deprecated
+  public void retain(int increment) {
+    referenceManager.retain(increment);
+  }
+
+  @Deprecated
+  public ArrowBuf clear() {
+    this.readerIndex = this.writerIndex = 0;
```

Comment:
agree this should be deprecated - readerIndex and writerIndex should move to NettyArrowBuf

Reply:
ideally we should not have duplicate methods doing the same work, at some point in the future we should remove the older..